mmv1/third_party/terraform/resources/resource_bigquery_table.go

```diff
@@ -441,9 +441,9 @@ func ResourceBigQueryTable() *schema.Resource {
 						"source_format": {
 							Type:        schema.TypeString,
 							Required:    true,
-							Description: `The data format. Supported values are: "CSV", "GOOGLE_SHEETS", "NEWLINE_DELIMITED_JSON", "AVRO", "PARQUET", "ORC" and "DATASTORE_BACKUP". To use "GOOGLE_SHEETS" the scopes must include "googleapis.com/auth/drive.readonly".`,
+							Description: `The data format. Supported values are: "CSV", "GOOGLE_SHEETS", "NEWLINE_DELIMITED_JSON", "AVRO", "ICEBERG", "PARQUET", "ORC" and "DATASTORE_BACKUP". To use "GOOGLE_SHEETS" the scopes must include "googleapis.com/auth/drive.readonly".`,
 							ValidateFunc: validation.StringInSlice([]string{
```

Comment:
```suggestion
```
Can be removed potentially -- see main comment

Reply:
This seems reasonable but I think we've typically tried to make clients more self documenting/provide useful error messages.  If you don't think this will degrade client experience I can make this change.

Reply:
Either way, I think we should probably make this change separately as it seems like it can have a larger blast radius.  I'll work on updating the referenced docs.

Reply:
The more direct error message that points to correct values via Terraform rather than generically pointing to the field in question as having a problem (aka the default API response to invalid requests) does come at a cost of maintainability of this resource -- if more options are enabled in the future it will similarly require a future PR to allow support. If there are no plans for future additions within the next year at least then it would be fine to keep it as an enum here but if there is any other potential additions in the near future, it would still be for the best to update the field as a whole.

Reply:
> If there are no plans for future additions within the next year at least then it would be fine to keep it as an enum here but if there is any other potential additions in the near future, it would still be for the best to update the field as a whole.

I think it depends on definition of near future.  But generally this is going to be a slow changing enum (i.e. maybe 1 addition per year).

Reply:
The api will fail with an appropriate value. Plan time linting is nice, but it incurs a maintenance burden. Values won't be available as they are supported by the API. It will take an engineer to preemptively make the change or a customer to file a bug against us to fix it. My personal preference is, consider the value a string with no validation. Supply a link to documentation of supported values. Allow the API to fail. Realistically if you are on the team your representation can take precedence here.

It appears you are a googler (by your knowledge). Could join the GoogleCloudOrganization so our CI can detect you? Join [GoogleCloudPlatform org](https://github.com/GoogleCloudPlatform) through [go/github/orgs](http://go/github/orgs)

https://github.com/orgs/GoogleCloudPlatform/people?query=emkornfield

Reply:
Based on the link, I think I am already a member (maybe something is off, it shows all lowercase for me).

I'm on the team that owns these formats, I'll bring it back to the team on whether we want to remove this going forward

Reply:
oh you are ! sorry, usually googlers have the little flair from https://chromewebstore.google.com/detail/dlebflppeeemcdpidccbiblndppbmjmh?pli=1